lib/matplotlib/cbook/__init__.py

```diff
@@ -1990,6 +1990,8 @@ def _warn_external(message, category=None):
     """
     frame = sys._getframe()
     for stacklevel in itertools.count(1):  # lgtm[py/unused-loop-variable]
+        if frame is None:
```

Comment:
Would suggest leaving a comment (as just below) so that people realize when this can happen.

Reply:
I took the liberty of adding such a comment